### PR TITLE
Update OMR README with Travis Matrix Builds

### PR DESCRIPTION
**Pull request by aarongraham9**
Travis: Update OMR README
-Add breakdown of Travis CI matrix builds to main OMR README
-Arrange Builds by Platform in OMR README

[skip ci]

Signed-off-by: Aaron Graham <aaron.graham@unb.ca>

### CODE REVIEW

File: compiler/arm/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# Eclipse OMR
+
+| Build | Status |
+| ---------------------- | -------------------- |
+| **Travis CI** | [![Travis Status](https://api.travis-ci.org/eclipse/omr.svg?branch=master)](https://travis-ci.org/eclipse/omr) |
+| **Travis CI Linux ARM Matrix Build** | [![Travis Status](https://travis-matrix-badges.herokuapp.com/repos/eclipse/omr/branches/master/4)](https://travis-ci.org/eclipse/omr) |
+ARM (32-Bit)
```
Comment (Leonardo2718) [inline compiler/arm/README.md]: An extra blank line is needed here. Otherwise, GitHub will interpret this as part of the table.

File: compiler/arm/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+====================
+
+The ARM port of OMR is ongoing and is a work in progress. If you have any ARM related questions or feedback please [open an Eclipse OMR GitHub issue][omrarmissue] or post in the [eclipse-omr Slack][omrslack].
```
Comment (Leonardo2718) [inline compiler/arm/README.md]: The phrase "is ongoing and is a work in progress" seems redundant. I would shorten the sentence to just "The ARM port of OMR is a work in progress"

File: compiler/aarch64/README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+| Build | Status |
+| ---------------------- | -------------------- |
+| **Travis CI** | [![Travis Status](https://api.travis-ci.org/eclipse/omr.svg?branch=master)](https://travis-ci.org/eclipse/omr) |
+| **Travis CI Linux AArch64 Matrix Build** | [![Travis Status](https://travis-matrix-badges.herokuapp.com/repos/eclipse/omr/branches/master/3)](https://travis-ci.org/eclipse/omr) |
+
+AArch64 (ARM 64-Bit)
+====================
+
+The AArch64 port of OMR is ongoing and is a work in progress. If you have any AArch64 related questions or feedback please [open an Eclipse OMR GitHub issue][omraarch64issue] or post in the [#aarch64 channel in the eclipse-omr Slack][omrslack].
```
Comment (Leonardo2718) [inline compiler/aarch64/README.md]: The phrase "is ongoing and is a work in progress" seems redundant. I would shorten the sentence to just "The AARCH64 port of OMR is a work in progress"

File: compiler/aarch64/README.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+We have been working on building up the generic compiler technology and tests in OMR. We are looking to move our port into [OpenJ9][openj9] in 2019.
```
Comment (Leonardo2718) [inline compiler/aarch64/README.md]: "We are looking to move our port into OpenJ9 in 2019" is this actually relevant to the OMR project? I think this comment would be more of interest to the OpenJ9 community.

File: README.md [hunk 5]
```diff
@@ -25,6 +25,12 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
...
+| **Travis CI Linux x86-64 CMAKE Makefile Matrix Build** | [![Travis Status](https://travis-matrix-badges.herokuapp.com/repos/eclipse/omr/branches/master/1)](https://travis-ci.org/eclipse/omr) |
+| **Travis CI Linux x86-64 Matrix Build** | [![Travis Status](https://travis-matrix-badges.herokuapp.com/repos/eclipse/omr/branches/master/2)](https://travis-ci.org/eclipse/omr) |
+| **Travis CI Linux AArch64 Matrix Build** | [![Travis Status](https://travis-matrix-badges.herokuapp.com/repos/eclipse/omr/branches/master/3)](https://travis-ci.org/eclipse/omr) |
+| **Travis CI Linux ARM Matrix Build** | [![Travis Status](https://travis-matrix-badges.herokuapp.com/repos/eclipse/omr/branches/master/4)](https://travis-ci.org/eclipse/omr) |
+| **Travis CI Linux x86-64 LINT Matrix Build** | [![Travis Status](https://travis-matrix-badges.herokuapp.com/repos/eclipse/omr/branches/master/5)](https://travis-ci.org/eclipse/omr) |
+| **Travis CI Linux x86-64 CMAKE Ninja Matrix Build** | [![Travis Status](https://travis-matrix-badges.herokuapp.com/repos/eclipse/omr/branches/master/6)](https://travis-ci.org/eclipse/omr) |
```
Comment (Leonardo2718) [inline README.md]: It looks like all these links are pointing to the same location. Are all of them really needed?

**Comment by jeanlego** [comment]
Should we use separate tables per CI ( travis, appveyor, jenkins)

File: README.md (hunk 5 above)
Comment (aarongraham9) [reply README.md]: Each build badge is pointing to a separate one of the 6 matrix build in Travis. If one out of the 6 matrix builds fails it will explicitly show the one that is failing. The Travis build badge we have now is more of a black box and doesn't indicate that only one/some matrix build/platform failed or which one in particular failed. The URLs that the build badges "click to" are all the main eclipse/omr [Travis page](https://travis-ci.org/eclipse/omr) as that would allow one to dig deeper into a failure.

File: compiler/aarch64/README.md (hunk 4 above)
Comment (aarongraham9) [reply compiler/aarch64/README.md]: Agreed; @0xdaryl suggested we include it here because the two projects are connected.

File: compiler/aarch64/README.md (hunk 3 above)
Comment (aarongraham9) [reply compiler/aarch64/README.md]: Agreed. Will fix up.

File: compiler/arm/README.md (hunk 2 above)
Comment (aarongraham9) [reply compiler/arm/README.md]: Agreed. Will fix up.

File: compiler/arm/README.md (hunk 1 above)
Comment (aarongraham9) [reply compiler/arm/README.md]: Ok. Will fix up.

**Comment by aarongraham9** [comment]
@jeanlego I like it split out into separate tables, or separate sections of the same table. @0xdaryl mentioned It might be good to have it in one table grouped by platform. I'm good with either format. Does anyone have a strong preference?

File: README.md (hunk 5 above)
Comment (Leonardo2718) [reply README.md]: I see. In that case, I think the approach suggested by @jeanlego is tidier.

**Comment by aarongraham9** [comment]
I updated the PR based on your feedback @Leonardo2718 and changed the table to be sorted by platform as @0xdaryl suggested. Do you still prefer @jeanlego's style over sorted by platform? Thoughts anyone else?

**Comment by aarongraham9** [comment]
@charliegracie any thoughts on this?

**Comment by aarongraham9** [comment]
If no one else (@Leonardo2718, @charliegracie, @0xdaryl, @jeanlego) has any comments then this should be good to merge at your earliest convenience.

**Comment by aarongraham9** [comment]
@0xdaryl

**Comment by 0xdaryl** [comment]
Apologies for the delay in responding.  I'm afraid I've had a change of heart on the AArch64-specific README since you first mentioned it to me now that I've given it more thought and seen a concrete example.  Thanks for putting a PR together that demonstrates what you had in mind.

I don't think a separate README for an architecture-under-implementation is warranted mainly because I don't think it should get special treatment above and beyond other OMR features under development that don't have the "luxury" of having their own directory in which to place a README file for feature-specific info.  I'd like us to be fair and consistent within the project.

The compiler technology uses issues for work needed to be done, so I believe the right way to expose work you want the community to help with is to create tasks and tag them appropriately with labels (like `Help Wanted`, `New Platform`), and be sure to have AArch64 in the abstract.  Someone conducting a simple filtered search should find these issues, and it's always up-to-date.  We already mention in the project README that `Help Wanted` tasks are there.

In terms of reporting the status of the work, I'm less convinced that a README is the right way to report this anyway.  I think the Slack channel, omr-dev mailing list, and the eventual OMR release notes are better options in my opinion.  And again, I don't think AArch64 should be any different than other OMR features under development.

Now, I can accept the commit that provides more details on the TravisCI builds in the project README.  If you remove the AArch64 README commit from this PR I can merge it.  I find that level of detail useful.

**Comment by aarongraham9** [comment]
Agreed. I'll drop the Arch specific READMEs, rebase and update the PR with just the modified main README.

**Comment by 0xdaryl** [comment]
A documentation only change.  No need to wait for AppVeyor build which launched for some reason.  Merging.
